File: app/src/main/java/com/code/yashladha/android_user/Portal/Adapter/LogsAdapter.java
package com.code.yashladha.android_user.Portal.Adapter;

import android.content.Context;
import android.support.annotation.LayoutRes;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.code.yashladha.android_user.Models.ProductSold;
import com.code.yashladha.android_user.R;

import java.util.ArrayList;

/**
 * Created by User on 10/24/2017.
 */

public class LogsAdapter extends ArrayAdapter<ProductSold> {

    public LogsAdapter(Context context, ArrayList<ProductSold> soldItems, int resource) {
        super(context, 0, soldItems);
    }

    public View getView(int position, View convertView, ViewGroup parent) {

        View pro_list_view = convertView;

        if (pro_list_view == null) {
            pro_list_view = LayoutInflater.from(getContext()).inflate(
                    R.layout.sold_item, parent, false);
        }
        ProductSold currentPro = getItem(position);

        TextView nameTextView = (TextView) pro_list_view.findViewById(R.id.pname);
        nameTextView.setText(currentPro.getP_name());

        TextView pDateTextView = (TextView) pro_list_view.findViewById(R.id.purchase_date);
        pDateTextView.setText(currentPro.getDate());

        TextView priceTextView = (TextView) pro_list_view.findViewById(R.id.price);
        priceTextView.setText(currentPro.getPrice());

        TextView dDateTextView = (TextView) pro_list_view.findViewById(R.id.deliver_date);

        ImageView iconView = (ImageView) pro_list_view.findViewById(R.id.image);

        iconView.setVisibility(View.VISIBLE);

    }
}

File: app/src/main/java/com/code/yashladha/android_user/Portal/Adapter/LogsAdapter.kt
package com.code.yashladha.android_user.Portal.Adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import com.code.yashladha.android_user.Models.Logs
import com.code.yashladha.android_user.R
import kotlinx.android.synthetic.main.log_item.view.*

/**
 * Created by yashladha on 19/11/17.
 * Log Adapter for showing the order placed Log
 */

class LogsAdapter(val logsList: ArrayList<Logs>, val context: Context) : RecyclerView.Adapter<LogsAdapter.ViewHolder>() {
    override fun onBindViewHolder(holder: ViewHolder?, position: Int): Unit = holder!!.bind(logsList[position])

    override fun onCreateViewHolder(parent: ViewGroup?, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.log_item, parent, false)
        return ViewHolder(view)
    }

    override fun getItemCount(): Int = logsList.size

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        fun bind(item: Logs) = with(itemView) {
            itemView.log_timestamp.text = item.timeStamp
            val productCountString = "No. of products ordered: " + item.items.size.toString()
            itemView.logs_products_number.text = productCountString

            itemView.setOnClickListener {
                Toast.makeText(context, item.timeStamp, Toast.LENGTH_SHORT).show()
            }
        }
    }

}

